### DISCUSSION TITLE
Настройка на Keenetic

### DISCUSSION THREAD

**Discussion by Murakumo-JP** [xtrime-ru/antizapret-vpn-docker#30] [Q&A] [answered]
Снова Здраствуйте, во первых большое спасибо @xtrime-ru за помощь в прошлом вопросе, я всё поднял и развернул и всё работает, хотел настроить теперь его на Keenetic версия прошивки стоит 4.1.7, настраивал всё по инструкций с ntc.party, но почему то работает через раз, происходят типа обрывы связи, сделал как указана в инструкций отключил у основного соединения "Игнорировать DNS".
Конфиг выглядит примерно так, подключение у меня не PPPoE, обычное IPoE.

> nobind
> client
> 
> remote 9x.2xx.1xx.1x
> 
> remote-cert-tls server
> 
> dev tun
> proto tcp
> cipher AES-128-CBC
> 
> resolv-retry infinite
> persist-key
> persist-tun
> 
> pull-filter ignore block-outside-dns
> route 8.8.8.8

Ютуб и всё такое открывает но грузит почему то через раз, и ещё вопрос можно ли как то это всё подружить с DoT/ DoH ? у меня проста была фильтрация от cloudflare.

**Comment by xtrime-ru** [comment]
А почему конфиг клиента отличается от того, что генерирует контейнер?

**Reply by Murakumo-JP** [reply]
Я вставлял сначала конфиг целиком но роутер его вообще не как не воспринимал и ничего не работала 
```
nobind
client

remote 9x.2xx.1xx.1x

remote-cert-tls server

dev tun
proto tcp
keepalive 2 10
tun-mtu 1420
sndbuf 3670016
rcvbuf 3670016
tcp-nodelay

resolv-retry infinite
persist-key
persist-tun

setenv FRIENDLY_NAME "AntiZapret VPN TCP"

# Keys
<ca>
-----BEGIN CERTIFICATE-----

-----END CERTIFICATE-----
</ca>

<cert>
-----BEGIN CERTIFICATE-----

-----END CERTIFICATE-----
</cert>

<key>
-----BEGIN PRIVATE KEY-----

-----END PRIVATE KEY-----
</key>
```

**Reply by xtrime-ru** [reply]
Да, это верный конфиг. Есть лог ошибок с роутера? 
Так же рекомендую попробовать udp версию. Она гораздо быстрее и меньше нагружает CPU сервера и клиента.

**Reply by xtrime-ru** [reply]
Все работает, клиент пишет: 

[I] Aug 15 13:28:30 OpenVPN0: Initialization Sequence Completed 

Никаких ошибок нет, есть предупреждения --cipher is not set и setting tcp-nodelay on the client side will not affect the server, но они не влияют на работу и у меня тоже есть.

**Reply by Murakumo-JP** [reply]
Хм.... странно тогда, почему то ютуб тот же вот сейчас снова грузить перестал, проста нет активности и всё.
![image](https://github.com/user-attachments/assets/7f4fea7d-7e73-4c88-9f7a-5a122208ed7a)

**Reply by xtrime-ru** [reply]
Что выдает команда?
```
 #(для tcp коннекта)
nslookup youtube.com 192.168.104.1

 #(для udp коннекта)
nslookup youtube.com 192.168.100.1
```

**Reply by Murakumo-JP** [reply]
При тесте DNS что очень странно выдаёт сначала моё место положения а потом уже моего VPN 
![image](https://github.com/user-attachments/assets/012fd73a-0592-4dce-b7f7-1596a406ddcd)

**Reply by xtrime-ru** [reply]
Нужны результаты команды nslookup или dig. Можно также traceroute youtube.com

Вот так у меня коннекте через роутер с стандартным upd  конфигом AZ: 

<img width="1006" alt="Снимок экрана 2024-08-15 в 12 48 18" src="https://github.com/user-attachments/assets/277d31dc-e222-4173-af7f-65bd506b04ec">

**Reply by xtrime-ru** [reply]
Все говорит о том, что коннекта нет. Что в статусе OpenVPN подключения в роутере?

Есть возможность сделать nsolookup на самом роутере?  Зайти на роутер по SSH или из панели управления роутера.

**Reply by Murakumo-JP** [reply]
> nslookup youtube.com 192.168.104.1

В статусе OpenVPN пишет что подключение есть, а вот как к роутеру по SSH подключится я не знаю извините я в этом полный нуб.

**Reply by Murakumo-JP** [reply]
В роутере через панель ток умею пинг и трасировку делать 
TCP
```
starting traceroute to 192.168.104.1...
traceroute to 192.168.104.1 (192.168.104.1), 30 hops maximum, 52 byte packets.
1 192.168.104.1 (192.168.104.1) 41.255 ms 44.832 ms 40.939 ms
```
UDP
```
starting traceroute to 192.168.100.1...
traceroute to 192.168.100.1 (192.168.100.1), 30 hops maximum, 52 byte packets.
1 192.168.100.1 (192.168.100.1) 41.373 ms 41.131 ms 40.995 ms
```
Это всё что я получаю.

**Reply by xtrime-ru** [reply]
А что говорит traceroute youtube.com в роутере?

То что оба трейроута сработали это плохо. У вас должен работать только до 192.168.104.1. До 192.168.100.1 должен идти через WAN.

По какой то причине роутер считает оба маршрута локальными. Возможно какой то конфликт настройк и вместо того, что бы идти в тунель через интернет он пытается идти в локальную сеть.

**Reply by xtrime-ru** [reply]
DNS неправильный, должен быть 10.244.0.0. 
Но штатные инструменты роутеров могут не использовать openvpn. У меня например asus через админку игнорирует DNS правила работающего openvpn. 

Надо разобраться почему traceroute до  192.168.100.1 работает.

**Reply by Murakumo-JP** [reply]
Хорошо попробую разобраться в этом, попробую сбросить роутер к заводским настройкам посмотреть как поведёт себя, извините что отнял у вас только времени и спасибо вам.

**Reply by crubik** [reply]
> Хорошо попробую разобраться в этом, попробую сбросить роутер к заводским настройкам посмотреть как поведёт себя, извините что отнял у вас только времени и спасибо вам.

рекомендую пул адресов за NAT сменить с 192.168.0.х на 10.0.0.х

**Reply by Murakumo-JP** [reply]
> > Хорошо попробую разобраться в этом, попробую сбросить роутер к заводским настройкам посмотреть как поведёт себя, извините что отнял у вас только времени и спасибо вам.
> 
> рекомендую пул адресов за NAT сменить с 192.168.0.х на 10.0.0.х

Извините я не очень в этом понимаю, суть в том что я не получаю ip динамически у меня выдан провайдером а за NAT я всегда привязан к одному ip (не белый ip), возможно мой провайдер чёт начудил по этому роутер и не понимает что делать я даже не знаю, по сути весь город у нас эта одна большая локальная сеть или как это назвать я даже не знаю.

**Reply by crubik** [reply]
> Извините я не очень в этом понимаю, суть в том что я не получаю ip динамически у меня выдан провайдером а за NAT я всегда привязан к одному ip (не белый ip), возможно мой провайдер чёт начудил по этому роутер и не понимает что делать я даже не знаю, по сути весь город у нас эта одна большая локальная сеть или как это назвать я даже не знаю.
https://help.keenetic.com/hc/ru/articles/360000512800-%D0%9A%D0%B0%D0%BA-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B8%D1%82%D1%8C-IP-%D0%B0%D0%B4%D1%80%D0%B5%D1%81-%D0%B8%D0%BD%D1%82%D0%B5%D1%80%D0%BD%D0%B5%D1%82-%D1%86%D0%B5%D0%BD%D1%82%D1%80%D0%B0-%D0%B2-%D0%B4%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D0%B5%D0%B9-%D1%81%D0%B5%D1%82%D0%B8-%D0%B8-%D0%BD%D0%B0%D1%81%D1%82%D1%80%D0%BE%D0%B9%D0%BA%D0%B8-DHCP

**Reply by xtrime-ru** [reply]
Да, стоит попробовать, но лучше выбрать 172.16.0.0, что бы с 10.244.0.0 совсем не пересекалось. https://ru.wikipedia.org/wiki/Частный_IP-адрес

**Reply by Murakumo-JP** [reply]
Сменил с 192.168.0.х на 172.16.0.0 и перезагрузил роутер, кажись это не работает для меня эх...
А ещё вечно в логе журнала вижу эту чёртову ошибку 
```
Io::File: unable to open "/proc/resetnds": no such file or directory.
```

**Reply by crubik** [reply]
> 172.16.0.0 и перезагрузил роутер, кажись это не работает для меня эх...

ПК какой ip получило?

**Reply by Murakumo-JP** [reply]
> > 172.16.0.0 и перезагрузил роутер, кажись это не работает для меня эх...
> 
> ПК какой ip получило?

172.16.0.42

**Reply by crubik** [reply]
> 172.16.0.42
покажите весь конфиг опенвпн + стоит ли галка, на соединении - получать удаленные маршруты?

**Reply by crubik** [reply]
> Список маршрутов выглядит вот так вообще ![image](https://private-user-images.githubusercontent.com/32967450/358213930-2a99de76-ab03-4f3d-9ee1-ffb1c4158d93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjQxMjMsIm5iZiI6MTcyMzcyMzgyMywicGF0aCI6Ii8zMjk2NzQ1MC8zNTgyMTM5MzAtMmE5OWRlNzYtYWIwMy00ZjNkLTllZTEtZmZiMWM0MTU4ZDkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDEyMTAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2MDEzZTMyZjdjZTkxZmQ2MGJjODgwNTUxNzZhMjQyZmQ3M2EyODFkNjA4NzJkYjliMzBkM2RkOTg5ZmViOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GqqNletdBd1__QTEeD4TQcpU_WhM5Pv2SlGZaSkGnFk)

стоп. а где 0.0.0.0/0  на интерфейсе провайдера? у вас интернет то работает?

**Reply by Murakumo-JP** [reply]
> > 172.16.0.42
> > покажите весь конфиг опенвпн + стоит ли галка, на соединении - получать удаленные маршруты?

Так же добавил туда DNS от провайдера которые, весь конфиг выше был.
```
pull-filter ignore block-outside-dns
route 10.10.10.7
route 8.8.8.8
```
![image](https://github.com/user-attachments/assets/1e0bdb82-8e9f-42c9-a684-e84ecd0488c5)

**Reply by Murakumo-JP** [reply]
> стоп. а где 0.0.0.0/0 на интерфейсе провайдера? у вас интернет то работает?

Да интернет работает у меня 0.0.0.0/0 так же и на гостевой сети есть и на интернете я не знаю как роутер так раздаёт шлюз

**Reply by crubik** [reply]
вот так у меня (ip впс и ключи скрыл) и настройки что выше я писал. всё работает

```
pull-filter ignore block-outside-dns
route 77.88.8.8
route 9.9.9.10
verb 5

nobind
client

remote x.x.x.x.x

remote-cert-tls server

dev tun
proto tcp
keepalive 2 10
tun-mtu 1420
sndbuf 3670016
rcvbuf 3670016
tcp-nodelay

resolv-retry infinite
persist-key
persist-tun

setenv FRIENDLY_NAME "AntiZapret VPN TCP"

# Keys
<ca>
-----BEGIN CERTIFICATE-----

-----END CERTIFICATE-----
</ca>

<cert>
-----BEGIN CERTIFICATE-----

-----END CERTIFICATE-----
</cert>

<key>
-----BEGIN PRIVATE KEY-----

-----END PRIVATE KEY-----
</key>
```

**Reply by xtrime-ru** [reply]
А udp на кинетиках работает?

**Reply by crubik** [reply]
> Да интернет работает у меня 0.0.0.0/0 так же и на гостевой сети есть и на интернете я не знаю как роутер так раздаёт шлюз

![Screenshot_11](https://github.com/user-attachments/assets/7d1d9adc-85d1-4e74-8a61-876b60c2e92e)
попробуй до второй беты обновить

**Reply by Murakumo-JP** [reply]
Эх.... видать знать проще будет поставить OpenVPN Connect  на пк проста, он же будет так же проста разблокировать то что заблокирована или весь трафик через себя пропускать ?

**Reply by Murakumo-JP** [reply]
> > Да интернет работает у меня 0.0.0.0/0 так же и на гостевой сети есть и на интернете я не знаю как роутер так раздаёт шлюз
> 
> ![Screenshot_11](https://private-user-images.githubusercontent.com/35901117/358218024-7d1d9adc-85d1-4e74-8a61-876b60c2e92e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjUyODUsIm5iZiI6MTcyMzcyNDk4NSwicGF0aCI6Ii8zNTkwMTExNy8zNTgyMTgwMjQtN2QxZDlhZGMtODVkMS00ZTc0LThhNjEtODc2YjYwYzJlOTJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDEyMjk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0NmYyZWU1NTU1MzdiYWNkNDRiNjk4ZTA1NTRjODZhNzk2ZDAxMGFlODQwZjg3MWU0ZGViZGJlZGQ3MDEyYzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._Nzr4-gcV425W2h_OnO9ud3beWhkIjYqC0eCOd-zEcc) попробуй до второй беты обновить

у меня была бета 2 я тестил на ней тоже не работает, там ещё и кнопку игнорировать DNS не понятно куда запихали.

**Reply by crubik** [reply]
> > > Да интернет работает у меня 0.0.0.0/0 так же и на гостевой сети есть и на интернете я не знаю как роутер так раздаёт шлюз
> > 
> > 
> > ![Screenshot_11](https://private-user-images.githubusercontent.com/35901117/358218024-7d1d9adc-85d1-4e74-8a61-876b60c2e92e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjUyODUsIm5iZiI6MTcyMzcyNDk4NSwicGF0aCI6Ii8zNTkwMTExNy8zNTgyMTgwMjQtN2QxZDlhZGMtODVkMS00ZTc0LThhNjEtODc2YjYwYzJlOTJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDEyMjk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0NmYyZWU1NTU1MzdiYWNkNDRiNjk4ZTA1NTRjODZhNzk2ZDAxMGFlODQwZjg3MWU0ZGViZGJlZGQ3MDEyYzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._Nzr4-gcV425W2h_OnO9ud3beWhkIjYqC0eCOd-zEcc) попробуй до второй беты обновить
> 
> у меня была бета 2 я тестил на ней тоже не работает, там ещё и кнопку игнорировать DNS не понятно куда запихали.
вроде всё там же)
![Screenshot_12](https://github.com/user-attachments/assets/92de8a7c-39bc-47be-a2f4-0d4b6581e6aa)

**Reply by xtrime-ru** [reply]
Можно купить ASUS, там гарантированно все работает без танцев с бубном. Добавляешь ovpn файлик и все :) 

Любой openvpn клиент будет пускать через себя только DNS запросы и трафик до заблокированных ресурсов. 

Я бы еще почитал тему про кинетик: https://ntc.party/t/обход-блокировок-в-россии-через-антизапрет-на-прошивке-keenetic/7768/81

**Reply by crubik** [reply]
> Можно купить ASUS, там гарантированно все работает без танцев с бубном. Добавляешь ovpn файлик и все :)

Асус после серии RT-N56U уже не тот)

**Reply by Murakumo-JP** [reply]
> > вроде всё там же)
> > ![Screenshot_12](https://private-user-images.githubusercontent.com/35901117/358218726-92de8a7c-39bc-47be-a2f4-0d4b6581e6aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjU0NzAsIm5iZiI6MTcyMzcyNTE3MCwicGF0aCI6Ii8zNTkwMTExNy8zNTgyMTg3MjYtOTJkZThhN2MtMzliYy00N2JlLWEyZjQtMGQ0YjY1ODFlNmFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDEyMzI1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMTgyZGE5NzZiMDE3OWJjMzhlZTVhMjYzOWU4MjI1ZDBhZTg5MjQ1YmJiODBhODNjZWI0ZjNmODllYzIyNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lGW8nzontnXrDVHqDuxvC22VTgrXMpavGUp2P4Mlf8E)

Эта чёртова магия кинетика )))
![image](https://github.com/user-attachments/assets/c8c9b8bf-c534-4910-a0cd-d43c33a38a6c)

**Reply by crubik** [reply, marked as answer]
вот тут и убрать днс, предварительно прописав их в интернет-фильтре
![image](https://github.com/user-attachments/assets/2129e523-c6e4-46c6-9c25-f3c99032cd05)

**Reply by Murakumo-JP** [reply]
> вот тут и убрать днс, предварительно прописав их в интернет-фильтре

ОООО СРАНЬ ГОСПОДНЯ ОНО ЗАРАБОТАЛА ЕМОЁ

**Reply by Murakumo-JP** [reply]
Спасибо @xtrime-ru и @crubik и извините что отнял у вас только времени.

**Reply by xtrime-ru** [reply]
![image](https://github.com/user-attachments/assets/813c055e-5f09-4315-a214-313e7b696abd)

**Reply by xtrime-ru** [reply]
@Murakumo-JP Попробуй плиз udp еще! 

На нем скорость должна быть сильно выше. Интересно работает оно на кинетиках или нет.

**Reply by crubik** [reply]
> ![image](https://private-user-images.githubusercontent.com/34161928/358221496-813c055e-5f09-4315-a214-313e7b696abd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjYyMjksIm5iZiI6MTcyMzcyNTkyOSwicGF0aCI6Ii8zNDE2MTkyOC8zNTgyMjE0OTYtODEzYzA1NWUtNWYwOS00MzE1LWEyMTQtMzEzZTdiNjk2YWJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDEyNDUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5OGUwYWJmOWRiYzU5NTZjZGQ1MTRhZWVjYTJhMTkwOGM0NDc0OGFiOGZlNzY1Yzc5NjQ1NmFjMzFkY2VlZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8nEXMToSR55bWckMOGREcZebXSNmolpMtNjwQk84VyM)

ну микротик с антизапретом сложнее, там докер внутри него поднимать нужно. Проще через WG и точечно на запрещенку)

**Reply by crubik** [reply]
> @Murakumo-JP Попробуй плиз udp еще!
> 
> На нем скорость должна быть сильно выше. Интересно работает оно на кинетиках или нет.

трафик вроде идет по удп. роутер просто дома) 
![Screenshot_13](https://github.com/user-attachments/assets/f200991c-9c25-43fc-8d2c-2446f68ef483)

**Reply by Murakumo-JP** [reply]
> @Murakumo-JP Попробуй плиз udp еще!
> 
> На нем скорость должна быть сильно выше. Интересно работает оно на кинетиках или нет.

да по udp тоже работает.

**Reply by MascotWorld** [reply]
> @Murakumo-JP Попробуй плиз udp еще!
> 
> На нем скорость должна быть сильно выше. Интересно работает оно на кинетиках или нет.

Я с января на кинетике сижу на удп, сейчас полностью обновил впн, работать перестало))) Чтобы настроить кинетик и  нужно найти гайд для общего antizapret и вставить свои ключи в конфиг и настроить роутер по гайду. У меня было так, но перестало работать, ниже crubik написал во втором пункте что добавить, я у себя поменял с comp-zlo на verb5 и все заработало

**Reply by MascotWorld** [reply]
> Можно купить ASUS, там гарантированно все работает без танцев с бубном. Добавляешь ovpn файлик и все :)
> 
> Любой openvpn клиент будет пускать через себя только DNS запросы и трафик до заблокированных ресурсов.
> 
> Я бы еще почитал тему про кинетик: [https://ntc.party/t/обход-блокировок-в-россии-через-антизапрет-на-прошивке-keenetic/7768/81](https://ntc.party/t/%D0%BE%D0%B1%D1%85%D0%BE%D0%B4-%D0%B1%D0%BB%D0%BE%D0%BA%D0%B8%D1%80%D0%BE%D0%B2%D0%BE%D0%BA-%D0%B2-%D1%80%D0%BE%D1%81%D1%81%D0%B8%D0%B8-%D1%87%D0%B5%D1%80%D0%B5%D0%B7-%D0%B0%D0%BD%D1%82%D0%B8%D0%B7%D0%B0%D0%BF%D1%80%D0%B5%D1%82-%D0%BD%D0%B0-%D0%BF%D1%80%D0%BE%D1%88%D0%B8%D0%B2%D0%BA%D0%B5-keenetic/7768/81)

У родителей асус, там да, можно добавить файлик ovpn, но оно не работает (от слова совсем), читал на форумах, на ранних прошивках оно работало, но потом перестало)

**Reply by xtrime-ru** [reply]
> поменял с comp-zlo на verb5 и все заработало

В данном контейнере отключена поддержка сжатия, так как она признана устаревшей и не позволяет серверу использовать DCO.

> У родителей асус, там да, можно добавить файлик ovpn, но оно не работает (от слова совсем)

У меня ASUS AX 55 c прошивкой 3.0.0.4.386_52303. Все работает. Какая версия у вас? Файл ovpn актуальный? 

Если прошивка старая, то возможно нет поддержи GCM шифрования. Нужно руками вернуть в data-chiphers старое шифрование AES-128-CBC на сервере. Это выключит DCO, но хотя бы будет работать. 

Можно лог коннекта? У меня выглядит так: 

```text
Aug 18 17:05:30 vpnclient5[14169]: NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Aug 18 17:05:30 vpnclient5[14169]: WARNING: normally if you use --mssfix and/or --fragment, you should also set --tun-mtu 1500 (currently it is 1420)
Aug 18 17:05:30 vpnclient5[14169]: TCP/UDP: Preserving recently used remote address: [AF_INET]XXXX:1194
Aug 18 17:05:30 vpnclient5[14169]: Socket Buffers: R=[524288->1048576] S=[524288->1048576]
Aug 18 17:05:30 vpnclient5[14169]: UDP link local: (not bound)
Aug 18 17:05:30 vpnclient5[14169]: UDP link remote: [AF_INET]XXXX:1194
Aug 18 17:05:40 vpnclient5[14169]: [UNDEF] Inactivity timeout (--ping-restart), restarting
Aug 18 17:05:40 vpnclient5[14169]: SIGUSR1[soft,ping-restart] received, process restarting
Aug 18 17:05:40 vpnclient5[14169]: Restart pause, 5 second(s)
Aug 18 17:05:45 vpnclient5[14169]: NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Aug 18 17:05:45 vpnclient5[14169]: WARNING: normally if you use --mssfix and/or --fragment, you should also set --tun-mtu 1500 (currently it is 1420)
Aug 18 17:05:45 vpnclient5[14169]: TCP/UDP: Preserving recently used remote address: [AF_INET]XXXXXX:1194
Aug 18 17:05:45 vpnclient5[14169]: Socket Buffers: R=[524288->1048576] S=[524288->1048576]
Aug 18 17:05:45 vpnclient5[14169]: UDP link local: (not bound)
Aug 18 17:05:45 vpnclient5[14169]: UDP link remote: [AF_INET]XXXXX:1194
Aug 18 17:05:46 vpnclient5[14169]: TLS: Initial packet from [AF_INET]XXXXXX:1194, sid=0facdfc8 d8921310
Aug 18 17:05:46 vpnclient5[14169]: VERIFY OK: depth=1, CN=AntiZapret CA
Aug 18 17:05:46 vpnclient5[14169]: VERIFY KU OK
Aug 18 17:05:46 vpnclient5[14169]: Validating certificate extended key usage
Aug 18 17:05:46 vpnclient5[14169]: ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Aug 18 17:05:46 vpnclient5[14169]: VERIFY EKU OK
Aug 18 17:05:46 vpnclient5[14169]: VERIFY OK: depth=0, CN=antizapret-server
Aug 18 17:05:46 vpnclient5[14169]: WARNING: 'cipher' is present in local config but missing in remote config, local='cipher BF-CBC'
Aug 18 17:05:46 vpnclient5[14169]: Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA
Aug 18 17:05:46 vpnclient5[14169]: [antizapret-server] Peer Connection Initiated with [AF_INET]XXXXXX:1194
Aug 18 17:05:47 vpnclient5[14169]: SENT CONTROL [antizapret-server]: 'PUSH_REQUEST' (status=1)
Aug 18 17:05:48 vpnclient5[14169]: PUSH: Received control message: 'PUSH_REPLY...
Aug 18 17:05:48 vpnclient5[14169]: Options error: Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:3: block-outside-dns (2.4.12)
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: timers and/or timeouts modified
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: --ifconfig/up options modified
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: route options modified
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: route-related options modified
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: peer-id set
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: adjusting link_mtu to 1544
Aug 18 17:05:48 vpnclient5[14169]: OPTIONS IMPORT: data channel crypto options modified
Aug 18 17:05:48 vpnclient5[14169]: Data Channel: using negotiated cipher 'AES-128-GCM'
Aug 18 17:05:48 vpnclient5[14169]: Outgoing Data Channel: Cipher 'AES-128-GCM' initialized with 128 bit key
Aug 18 17:05:48 vpnclient5[14169]: Incoming Data Channel: Cipher 'AES-128-GCM' initialized with 128 bit key
Aug 18 17:05:48 vpnclient5[14169]: Preserving previous TUN/TAP instance: tun15
Aug 18 17:05:48 vpnclient5[14169]: NOTE: Pulled options changed on restart, will need to close and reopen TUN/TAP device.
Aug 18 17:05:48 vpnclient5[14169]: /etc/openvpn/ovpn-route-pre-down tun15 1420 1472 192.168.100.2 255.255.252.0 init
Aug 18 17:05:48 vpnclient5[14169]: Closing TUN/TAP interface
Aug 18 17:05:48 vpnclient5[14169]: /sbin/ifconfig tun15 0.0.0.0
Aug 18 17:05:48 vpnclient5[14169]: /etc/openvpn/ovpn-down tun15 1420 1472 192.168.100.2 255.255.252.0 init
Aug 18 17:05:49 vpnclient5[14169]: TUN/TAP device tun15 opened
Aug 18 17:05:49 vpnclient5[14169]: TUN/TAP TX queue length set to 100
Aug 18 17:05:49 vpnclient5[14169]: /sbin/ifconfig tun15 192.168.100.5 netmask 255.255.252.0 mtu 1420 broadcast 192.168.103.255
Aug 18 17:05:49 vpnclient5[14169]: /etc/openvpn/ovpn-up tun15 1420 1472 192.168.100.5 255.255.252.0 init
Aug 18 17:05:49 vpnclient5[14169]: Initialization Sequence Completed
```

**Comment by crubik** [comment]
Приветствую. DoH и DoT можно, но они будут не через антизапрет проходить
1. в компонентах кинетика установи **Фильтрация контента и блокировка рекламы при помощи облачных сервисов**
2. в конфигурации соединения опенвпн добавь строки 
pull-filter ignore block-outside-dns
route 77.88.8.8
route 9.9.9.10
verb 5
3. отключи получение днс от провайдера
![Screenshot_2](https://github.com/user-attachments/assets/af641a86-f814-4851-8cf1-f09577377450)

4. Интернет-фильтры - Настройка DNS - Добавить профиль - AntiZapret (галочка Транзитные запросы блокируются). Системный - Разрешен транзит запросов 
5. в Профиль AntiZapret добавить два DNS сервера что ты указал в конфиг опенвпн - 77.88.8.8 и 9.9.9.10
![Screenshot_8](https://github.com/user-attachments/assets/132e4f50-ef7c-47c9-ba46-e41eaf10887e)

6. в Профиль Системный добавить два-три DoH фильтра, которые не пересекаются с профилем AntiZapret. У меня это https://dns.google/dns-query и https://cloudflare-dns.com/dns-query 
![Screenshot_9](https://github.com/user-attachments/assets/06261af0-6282-4aae-9eca-399215253f61)

7. Интернет-фильтры - Режим фильтрации - Публичные DNS-резолверы - Профили контентной фильтрации по умолчанию - Antizapret
![Screenshot_10](https://github.com/user-attachments/assets/771f9dfb-2f8d-445b-a376-9b1cf1371962)

**Reply by Murakumo-JP** [reply]
Спасибо попробую настроить.

**Reply by xtrime-ru** [reply]
Зачем делать так , что бы рандомные домены не шли через антизапрет? Настоятельно не рекомендую так делать.

**Reply by Murakumo-JP** [reply]
Ну я пока всё это по удалял у себя так как да на форуме не советовали такое творить.

**Reply by crubik** [reply]
> Зачем делать так , что бы рандомные домены не шли через антизапрет? Настоятельно не рекомендую так делать.

не рандомные, а только те, что попадают в фильтр блокировки. или я что-то не так сделал - поправьте, если не трудно

**Reply by xtrime-ru** [reply]
Возможно, был не прав. В теме https://ntc.party/t/обход-блокировок-в-россии-через-антизапрет-на-прошивке-keenetic/7768/81  так же советуют добавлять route.

**Reply by crubik** [reply]
> советуют добавлять route.

так он у меня есть в п.2 инструкции выше

**Reply by xtrime-ru** [reply]
Да, поэтому написал, что был не прав и это не должно влиять на работоспособность.

**Comment by shilophrenic** [comment]
Возможно это не связанно с кинетиком, но как факт - настроен vpn с контейнером, телек прекрасно грузит ютуб, а вот в телефонах (в приложении) долго прогружает контент. Сталкивался кто ?

**Reply by Murakumo-JP** [reply]
Я столкнулся ток с проблемой что на мобильном интернете от Теле2 идёт вечный дисконект к серверу с vpn, в домашней сети проблем пока не наблюдаю всё грузит прекрасно.

**Comment by Murakumo-JP** [comment]


**Reply by xtrime-ru** [reply]
TCP: connect to [AF_INET]9x.2xx.1xx.1x:1194 failed: Host is unreachable  
Нет коннекта до сервера.

**Reply by Murakumo-JP** [reply]
Хотя сервер работает что очень странно
